roles/local-build/tasks/main.yml

```diff
@@ -1,92 +1,107 @@
-  ansible.builtin.set_fact:
-    kaia_build_required: True
-  when:
-    - kaia_build_check_file_stat.stat.exists == False or 
-      kaia_build_skip_if_exists == False
+    path: "{{ kaia_build_check_file_path }}-{{ build_identifiers[item] }}"
+  register: kaia_build_check_file_stats
+  loop: "{{ range(0, total_builds | int) | list }}"
 
 - name: Build - Clean the old directory
   ansible.builtin.file:
     state: absent
-    path: "{{ kaia_build_dir }}"
-  when:
-    - kaia_build_required
+    path: "{{ kaia_build_dir }}-{{ build_identifiers[item] }}"
+  when: not kaia_build_check_file_stats.results[item].stat.exists or not kaia_build_skip_if_exists or kaia_build_required
+  loop: "{{ range(0, total_builds | int) | list }}"
 
 - name: Build - Git checkout
   ansible.builtin.git:
-    repo: "{{ kaia_build_remote_git_url }}"
-    dest: "{{ kaia_build_dir }}"
-    version: "{{ kaia_build_remote_git_branch }}"
-  when:
-    - kaia_build_required
+    repo: "{{ git_url[item % git_url | length] if git_url is iterable and git_url is not string else git_url }}"
+    dest: "{{ kaia_build_dir }}-{{ build_identifiers[item] }}"
+    version: "{{ git_branch[item % git_branch | length] if git_branch is iterable and git_branch is not string else git_branch }}"
+  loop: "{{ range(0, total_builds | int) | list }}"
+
+- name: Build - Create bin directory
+  ansible.builtin.file:
+    path: "{{ bin_dir }}-{{ build_identifiers[item] }}"
+    state: directory
+  when: not kaia_build_check_file_stats.results[item].stat.exists or not kaia_build_skip_if_exists or kaia_build_required
+  loop: "{{ range(0, total_builds | int) | list }}"
 
 - name: Build - Compile kaia binary
   ansible.builtin.command: |
-    docker build --build-arg KLAYTN_STATIC_LINK=1 --build-arg DOCKER_BASE_IMAGE={{ kaia_build_docker_base_image }} -t {{ kaia_build_docker_builder_image }} --output output .
+    docker build --build-arg KLAYTN_STATIC_LINK=1 --build-arg DOCKER_BASE_IMAGE={{ kaia_build_docker_base_image }} -t {{ kaia_build_docker_builder_image }}-{{ item }} --output output .
   args:
-    chdir: "{{ kaia_build_dir }}"
+    chdir: "{{ kaia_build_dir }}-{{ build_identifiers[item] }}"
   environment:
     DOCKER_BUILDKIT: 1
-  when:
-    - kaia_build_required
+  when: not kaia_build_check_file_stats.results[item].stat.exists or not kaia_build_skip_if_exists or kaia_build_required
+  loop: "{{ range(0, total_builds | int) | list }}"
+
+- name: Build - Copy kaia binary files
+  ansible.builtin.copy:
+    dest: "{{ bin_dir }}-{{ build_identifiers[item[0]] }}"
+    src: "{{ kaia_build_dir }}-{{ build_identifiers[item[0]] }}/output/klaytn-docker-pkg/bin/{{ item[1] }}"
+    mode: preserve
+  when: not kaia_build_check_file_stats.results[item[0]].stat.exists or not kaia_build_skip_if_exists or kaia_build_required
+  with_nested:
+    - "{{ range(0, total_builds | int) | list }}"
+    - ['kbn', 'kcn', 'kpn', 'ken', 'kscn', 'kspn', 'ksen']
 
+# We need to copy homi binary for the first build only
```

Comment:
Could you write the reason here and also describe them on "README.md"?

Reply:
Done, thank you for the suggestions.